### PR TITLE
Fixed a permissions issue when using `final-dev` containers 

### PR DESCRIPTION
**Pull request by jathanism**
# Closes: #4300 
# What's Changed

This addresses potential filesystem permissions issues on mounted volumes for Nautobot App developers using Linux native Docker workflows. 

A `USER root` statement is added to the `final-dev` build stage before the `EXPOSE  8080` command identically to how it was solved for the `dev stage.

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

File: changes/4300.fixed [hunk 1]
```diff
@@ -0,0 +1 @@
+Fixed a permissions issue when using `final-dev` containers by  switching to root user before exposing port and entrypoint.
```
Comment (HanlinMiao) [inline changes/4300.fixed:1]:
```suggestion
Fixed a permission issue when using `final-dev` containers by switching to root user before exposing port and entrypoint.
```

**Review by HanlinMiao — APPROVED** [review]

**Review by gsnider2195 — APPROVED** [review]

**Review by timizuoebideri1 — APPROVED** [review]

File: docker/Dockerfile [hunk 2]
```diff
@@ -245,6 +245,10 @@ ENV NAUTOBOT_INSTALLATION_METRICS_ENABLED=false
 RUN nautobot-server init && \
     nautobot-server build_ui
 
+# switch to root user for final-dev stage: https://github.com/nautobot/nautobot/issues/4300
+# hadolint ignore=DL3002
+USER root
+
```
Comment (bryanculver) [inline docker/Dockerfile:251]: Are there any concerns/issues with these commands here: [https://github.com/nautobot/nautobot-plugin-golden-config/blob/develop/development/Dockerfile#L72-L76 being run](https://github.com/nautobot/nautobot-plugin-golden-config/blob/9cc0dbbb5d54c74dce539cc7ae986b1e0dd6b2b6/development/Dockerfile#L72-L76) being run as a different user than the `pip` command installing Nautobot and other packages above?

Comment (gsnider2195) [reply docker/Dockerfile:251]: Looks like we'll be fine. We're using `ENV VIRTUAL_ENV="/opt/nautobot"` to tell poetry and pip to install python packages to `/opt/nautobot` so even as root when you run a `pip install xyz` in the `final-dev` image, it installs to `/opt/nautobot`

Comment (jathanism) [reply docker/Dockerfile:251]: @bryanculver This was something we explicitly tested as well FYI, which was building the `Dockerfile` you linked from the revised `final-dev` container. It was a win.

**Pull request merged**